src/emulator.c

```diff
@@ -1326,6 +1520,12 @@ void ofono_emulator_set_indicator(struct ofono_emulator *em,
 	gboolean call;
 	gboolean callsetup;
 	gboolean waiting;
+	struct ofono_emulator *em = __ofono_atom_get_data(atom);
+
+	if (!valid_indication(em, atom, name))
```

Comment:
If someone calls ofono_emulator_set_indicator he already knows that the modem supports the OFONO_ATOM_TYPE_EMULATOR_HFP atom and we don't have to check that again. See src/network.c line 1380